### PR TITLE
ISPN-6068 Upgrade to JGroups 3.6.7.Final

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6068

* Use TCP_NIO2 in the test suite.
* Remove the log4j2 workaround in CommandAwareRpcDispatcher

The test suite seems to much faster with the new version of TCP_NIO2 (and about the same speed as with regular TCP).

### CODE REVIEW

File: core/src/test/resources/stacks/tcp.xml [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
          port_range="30"
-         recv_buf_size="20m"
+         recv_buf_size="2m"
          send_buf_size="640k"
          max_bundle_size="31k"
          use_send_queues="true"
          enable_diagnostics="false"
          tcp_nodelay="true"
-         bundler_type="sender-sends-with-timer"
+         bundler_type="transfer-queue"
```
Comment (pruivo) [inline core/src/test/resources/stacks/tcp.xml]: isn't ```sender-sends-with-timer``` better?

Comment (danberindei) [inline core/src/test/resources/stacks/tcp.xml]: Oops, I need to revisit these config changes... I had the TCP_NIO2 changes stashed in a local branch for a long time :)

Comment (danberindei) [inline core/src/test/resources/stacks/tcp.xml]: I removed the extra changes.

**Comment by pruivo** [comment]
pushing...

**Comment by pruivo** [comment]
integrated! thanks @danberindei !

**Pull request closed without merging**